File: src/com/mrpowergamerbr/loritta/utils/music/GuildMusicManager.java
package com.mrpowergamerbr.loritta.utils.music;

import com.sedmelluq.discord.lavaplayer.player.AudioPlayer;
import com.sedmelluq.discord.lavaplayer.player.AudioPlayerManager;

import net.dv8tion.jda.core.entities.Guild;

/**
 * Holder for both the player and a track scheduler for one guild.
 */
public class GuildMusicManager {
	/**
	 * Audio player for the guild.
	 */
	public final AudioPlayer player;
	/**
	 * Track scheduler for the player.
	 */
	public final TrackScheduler scheduler;

	/**
	 * Creates a player and a track scheduler.
	 * @param manager Audio player manager to use for creating the player.
	 */
	public GuildMusicManager(Guild guild, AudioPlayerManager manager) {
		player = manager.createPlayer();
		scheduler = new TrackScheduler(guild, player);
		player.addListener(scheduler);
	}

	/**
	 * @return Wrapper around AudioPlayer to use it as an AudioSendHandler.
	 */
	public AudioPlayerSendHandler getSendHandler() {
		return new AudioPlayerSendHandler(player);
	}
}

File: src/main/java/com/mrpowergamerbr/loritta/audio/GuildMusicManager.kt
package com.mrpowergamerbr.loritta.audio

import com.mrpowergamerbr.loritta.utils.loritta
import lavalink.client.player.LavalinkPlayer
import net.dv8tion.jda.core.entities.Guild

/**
 * Holder for both the player and a track scheduler for one guild.
 */
class GuildMusicManager(val guild: Guild) {
	/**
	 * Audio player for the guild.
	 */
	val player: LavalinkPlayer
			get() = loritta.audioManager.lavalink.getLink(guild).player

	/**
	 * Track scheduler for the player.
	 */
	val scheduler = TrackScheduler(guild, player)

	init {
		player.addListener(scheduler)
	}
}